client/landing/gutenboarding/onboarding-block/edit.tsx

```diff
@@ -2,24 +2,35 @@
  * External dependencies
  */
 import { __ as NO__ } from '@wordpress/i18n';
-import { SelectControl } from '@wordpress/components';
+import { Button } from '@wordpress/components';
+import { map } from 'lodash';
 import { useDispatch, useSelect } from '@wordpress/data';
 import React, { useCallback } from 'react';
 
 /**
  * Internal dependencies
  */
-import { SiteType } from '../store/types';
+import { SiteType, UNKNOWN_FORM_VALUE } from '../store/types';
 import { STORE_KEY } from '../store';
 import './style.scss';
 
 export default function OnboardingEdit() {
 	const { siteTitle, siteType } = useSelect( select => select( STORE_KEY ).getState() );
-	const { setSiteType, setSiteTitle } = useDispatch( STORE_KEY );
-	const handleTitleChange = useCallback(
+	const { resetSiteType, setSiteType, setSiteTitle } = useDispatch( STORE_KEY );
+	const updateTitle = useCallback(
 		( e: React.ChangeEvent< HTMLInputElement > ) => setSiteTitle( e.target.value ),
 		[ setSiteTitle ]
 	);
+	const updateSiteType = useCallback(
+		( e: React.ChangeEvent< HTMLInputElement > ) => setSiteType( e.target.value as SiteType ),
+		[ setSiteType ]
+	);
+
+	const siteTypeOptions = {
+		[ SiteType.BLOG ]: NO__( 'with a blog.' ),
+		[ SiteType.STORE ]: NO__( 'for a store.' ),
+		[ SiteType.STORY ]: NO__( 'to write a story.' ),
+	};
```

Comment:
We can use Typescript here to check if all `SiteType` options are covered. 
And also a function would prevent creating an object with each re-render.
```
function getSiteTypeLabel(type: SiteType): string {
    switch (type) {
        case SiteType.BLOG:
        case SiteType.STORE:
        case SiteType.STORY:
            return 'b';    
    }
}
```

Reply:
> We can use Typescript here to check if all `SiteType` options are covered. 

I think my TypeScript-fu isn't good enough to know how to do this :sweat_smile: Can you give me some pointers here?

Reply:
Moving `siteTypeOptions` out of the `render` for starters: af873f9a2c

Reply:
Ooops, see https://github.com/Automattic/wp-calypso/pull/37248#discussion_r342044733